src/hotspot/share/cds/filemap.cpp

```diff
@@ -1021,96 +1051,139 @@ bool FileMapInfo::check_archive(const char* archive_name, bool is_static) {
     return false;
   }
 
-  size_t sz = is_static ? sizeof(FileMapHeader) : sizeof(DynamicArchiveHeader);
-  void* header = os::malloc(sz, mtInternal);
-  memset(header, 0, sz);
-  size_t n = os::read(fd, header, (unsigned int)sz);
+  size_t sz = sizeof(GenericCDSFileMapHeader);
+  GenericCDSFileMapHeader* header = (GenericCDSFileMapHeader*)os::malloc(sz, mtInternal);
+  memset((void*)header, 0, sz);
+  size_t n = os::read(fd, (void*)header, (unsigned int)sz);
```

Comment:
There are currently 3 different places where we read the GenericCDSFileMapHeader, find out the size, and then read the real header.

- FileMapInfo::get_base_archive_name_from_header
- FileMapInfo::check_archive
- FileMapInfo::init_from_file

I think these should be consolidated into a single function.